apps/data/src/tasks/data_export_task.py

```
@@ -191,6 +200,14 @@ def export_data(df):
             dbutils.fs.put(f"{OUTPUT_PATH}/_SUCCESS", "", overwrite=True)
         elif FORMAT == "parquet":
             df.coalesce(1).write.mode("overwrite").parquet(OUTPUT_PATH)
+        elif FORMAT == "xlsx":
+            if row_count >= EXCEL_MAX_ROWS:
+                raise ValueError(
+                    f"Export has {row_count} rows, which exceeds Excel's {EXCEL_MAX_ROWS}-row "
+                    "limit per sheet. Use CSV or Parquet for datasets this large."
+                )
```

Ruff (0.15.20)

[warning] Abstract `raise` to an inner function

(TRY301)